pkg/streamingpromql/operators/aggregation.go

```diff
@@ -152,30 +155,67 @@ func (a *Aggregation) NextSeries(ctx context.Context) (types.InstantVectorSeries
 		thisSeriesGroup := a.remainingInnerSeriesToGroup[0]
 		a.remainingInnerSeriesToGroup = a.remainingInnerSeriesToGroup[1:]
 
-		if thisSeriesGroup.sums == nil {
+		if len(s.Floats) > 0 && thisSeriesGroup.floatSums == nil {
 			// First series for this group, populate it.
 
-			thisSeriesGroup.sums, err = a.Pool.GetFloatSlice(steps)
+			thisSeriesGroup.floatSums, err = a.Pool.GetFloatSlice(steps)
 			if err != nil {
 				return types.InstantVectorSeriesData{}, err
 			}
 
-			thisSeriesGroup.present, err = a.Pool.GetBoolSlice(steps)
+			thisSeriesGroup.floatPresent, err = a.Pool.GetBoolSlice(steps)
 			if err != nil {
 				return types.InstantVectorSeriesData{}, err
 			}
+			thisSeriesGroup.floatSums = thisSeriesGroup.floatSums[:steps]
+			thisSeriesGroup.floatPresent = thisSeriesGroup.floatPresent[:steps]
+		}
 
-			thisSeriesGroup.sums = thisSeriesGroup.sums[:steps]
-			thisSeriesGroup.present = thisSeriesGroup.present[:steps]
+		if len(s.Histograms) > 0 && thisSeriesGroup.histogramSums == nil {
+			// First series for this group, populate it.
+			thisSeriesGroup.histogramSums, err = a.Pool.GetHistogramPointerSlice(steps)
+			if err != nil {
+				return types.InstantVectorSeriesData{}, err
+			}
+			thisSeriesGroup.histogramSums = thisSeriesGroup.histogramSums[:steps]
 		}
 
 		for _, p := range s.Floats {
 			idx := (p.T - start) / interval
-			thisSeriesGroup.sums[idx] += p.F
-			thisSeriesGroup.present[idx] = true
+
+			// If a mix of histogram samples and float samples, the corresponding vector element is removed from the output vector entirely.
+			if thisSeriesGroup.histogramSums != nil && thisSeriesGroup.histogramSums[idx] != nil {
+				thisSeriesGroup.histogramSums[idx] = nil
+				continue
+			}
```

Comment:
Rather than doing this check here and in the `s.Histograms` loop below, how about we add a check for this just before we compute `floatPointCount`? Then we can skip all these checks unless we know we have both floats and histograms for the series.

I'm imagining something like:

```golang
if len(thisGroup.floatPresent) > 0 && len(thisGroup.histogramSums) > 0 {
  for idx, present := range thisGroup.floatPresent {
    if present && thisGroup.histogramSums[idx] != nil {
      thisGroup.floatPresent[idx] = false
      thisGroup.histogramSums[idx] = nil
      thisGroup.histogramPointCount--
    }
  }
}
```